### PR TITLE
Add autotest for key verification and fix several edge-cases

### PR DESCRIPTION
**Pull request by TobiasFella**

### CODE REVIEW

**Review by CarlSchwan — APPROVED** [review]
:heart:

File: lib/keyverificationsession.h [hunk 1]
```diff
@@ -19,6 +19,11 @@ struct QUOTIENT_API EmojiEntry {
     Q_GADGET
     Q_PROPERTY(QString emoji MEMBER emoji CONSTANT)
     Q_PROPERTY(QString description MEMBER description CONSTANT)
+
+public:
+    bool operator==(const EmojiEntry& rhs) const {
+        return emoji == rhs.emoji && description == rhs.description;
+    }
```
Comment (KitsuneRal) [inline lib/keyverificationsession.h]:
```suggestion
    bool operator==(const EmojiEntry& rhs) const = default;
```

File: autotests/testkeyverification.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+class TestKeyVerificationSession : public QObject
+{
+    Q_OBJECT
+
+private Q_SLOTS:
+    void testVerification();
+
+};
+
```
Comment (KitsuneRal) [inline autotests/testkeyverification.h]: You don't need to add a separate `.h` file here; instead, just define the class in `testkeyverification.cpp` and `#include "testverification.moc"` at the end of it (moc will auto-generate it) - see how it's done in `quotest.cpp`.

**Review by KitsuneRal — CHANGES_REQUESTED** [review]
I trust you with the logic but suggest a couple of simplifications.

**Review by KitsuneRal — APPROVED** [review]

**Pull request merged**